### DISCUSSION TITLE
Tree volume values seem unusually small

### DISCUSSION THREAD

**Discussion by MadPadCF** [csiro-robotics/treetools#5] [General]
We have a treeinfo output file, and have looked through the logic in treeinfo.cpp to double check how volume is calculated; but we've noticed that the calculated total volume (and by extension, the calculated volume of the stems) seem quite lower than what would be expected by the height and DBH/largest diameter values. 

I'm also curious how rounding of such small values is intended to work.

We're regularly finding trees ~2 m tall with around 3e-4 m^3 of volume; this seems oddly low. Even one of our larger trees, at 11 metres tall with a 36cm first stem diameter has a volume of just one cubic metre. If we simplify it as a cylinder, we'd expect to see about 4 times that. 

Are these the types of numbers we should be expecting? 

Additionally, I've noticed that, as the first `volume` field is the volume for the whole tree and is the sum of the remaining volumes, it seems to round values below 0.0001 to zero. Is that intended?

**Comment by thomas-lowe** [comment]
Hi, 
 all of the maths inside treetools is with doubles but when saving to the .txt files values are rounded to the nearest 0.0001, this is because these tree files are stored in text and can get very large and unweildy on large forests if every measurement has more decimal places. It is rounding to the nearest hundredth of a millimetre, which is well below the precision of most lidars, so shouldn't be a significant cause of inaccuracy. 

I have looked through the tree_info.cpp code and the calculation looks fine. If anything it should slightly overestimate tree volume as cylinder intersections at bifurcations are counted twice. 

Your example of 11 metres tall 36cm diameter should indeed be about 1m^3 if it were a single cylinder. Unless by 36cm you meant radius.

Underestimation can occur in rayextract trees for a two reasons:
1. --global_taper_factor is high (e.g. a very difficult scan) and --global_taper is set too low. This will make the trees thinner than they really are.
2. --crop_length is too high. For instance on very small trees a 1m default crop length would cut off the last metre of branches.

Hope this helps :)

**Reply by MadPadCF** [reply]
Thank you, forgot to divide by 2 in my volume calculation. The values are more consistent now. 

We'll have a look at tweaking those parameters, and avoiding working directly with the segment-level volume estimates to avoid rounding errors in the meantime.
